### DISCUSSION TITLE
Accessing Pyinstaller bootloaded environment from spawned process

### DISCUSSION THREAD

**Discussion by JoeZiminski** [pyinstaller/pyinstaller#8318] [Help]
Thanks for all your work on this excellent installer.

We are trying to package a python application which spawns a terminal emulator ([wezterm](https://wezfurlong.org/wezterm/index.html)) and runs python from within it. The reason we are doing this is because we use [textual](https://textual.textualize.io/) for a terminal-based GUI, and not all terminals render correctly (e.g. `cmd.exe`).

Fundamentally my question is: How can a terminal spawned in a subprocess inherit the bootloaded environment such that commands in the terminal (e.g. `python`) work as expected.

A MRE is below, run simply within `pyinstaller run_app.py` (and then, for now, manually copying the folder `_vendored` containing the `wezterm` executable).

I was hoping based on [this part of the documentation](https://pyinstaller.org/en/v6.4.0/common-issues-and-pitfalls.html#windows) that when spawning a new terminal, it might inherit the bootloaded environment. Unfortunately this does not seem to be the case, for example executing `python` (or versioned variants of) is not recognised, similar to a terminal when opened in the usual windows environment through the windows GUI. 

In this case, maybe it is necessary to perform these steps that pyinstaller performs in a new process (from this [doc](https://www.pyinstaller.org/en/stable/advanced-topics.html#the-bootstrap-process-in-detail)) to setup the environment?:

```
On Windows set the [activation context](http://msdn.microsoft.com/en-us/library/windows/desktop/aa374153(v=vs.85).aspx).

Load the Python dynamic library. The name of the dynamic library is embedded in the executable file.

Initialize Python interpreter: set sys.path, sys.prefix, sys.executable.
```
Alternatively, it would be awesome if it was possible to specify which console is used by pyinstaller when the `--console` flag is set to `True`. Then we could run our python code directly in a terminal we know can support `textual`.

MRE:
```
import os
import subprocess

dir_path = os.path.dirname(os.path.realpath(__file__))
path_to_wezterm = "_vendored/wezterm/windows/wezterm.exe"

subprocess.Popen(
    f"{dir_path}/{path_to_wezterm}"
)
```

Thanks!

**Comment by bwoodsend** [comment]
Hmm, this whole _bootloaded environment_ is a construct you've made up I'm afraid. A PyInstaller application is not a relocatable Python environment so no, it does not contain a `python` executable. If you're looking for an embedded Python then I suggest you use an embedded Python rather than PyInstaller.

**Comment by rokm** [comment]
As Brénainn already pointed out, PyInstaller-frozen application uses embedded python interpreter to run your application code, so you cannot use it from the outside of the application process.

> Alternatively, it would be awesome if it was possible to specify which console is used by pyinstaller when the `--console` flag is set to `True`. Then we could run our python code directly in a terminal we know can support `textual`.

This is up to OS to decide, not the application (i.e., contemporary versions of Windows allow you to globally switch between Windows Terminal and old `cmd.exe` console for console applications). The best you can probably do is to set up a launcher shortcut that opens/runs `wezterminal` and instructs it to run your program executable; probably something like `path/to/wezter.exe /start <path-to-program-exe>`?).
